test/benchmark/puppet_fact_importer_benchmark.rb

```diff
@@ -0,0 +1,51 @@
+require "benchmark/benchmark_helper"
+
+(0..10000).each do |x|
+  FactName.create!(:name => "random_fact_name_#{x}")
+end
+
+foreman_benchmark do
+  Benchmark.ips do |x|
+    #x.config(:time => 15, :warmup => 2)
```

Comment:
Leftover, actually this is the correct setting and the other one (1/1) was just for testing the benchmark itself. I see you executed the tests with 1/1, can you try again with 15/2? That should give better results. Also, are you testing with MySQL?